src/lib/network/clients/inst_lan.rb

```diff
@@ -59,20 +59,20 @@ def initialize
     end
 
     def main
+      ret = nil
       log.info("----------------------------------------")
       log.info("Lan module started")
 
       manual_conf_request = GetInstArgs.argmap["skip_detection"] || false
       log.info("Lan module forces manual configuration: #{manual_conf_request}")
 
-      log.info("Configured network found: #{network_configured?}")
-
-      ret = if network_configured? && !manual_conf_request
-        GetInstArgs.going_back ? :back : :next
-      else
-        LanSequence()
+      if !manual_conf_request
+        log.info("Configured network found: #{network_configured?}")
+        ret = GetInstArgs.going_back ? :back : :next if network_configured?
```

Comment:
should work also with simple
```suggestion
        ret = :auto if network_configured?
```

Reply:
So, what about:

```ruby
      ret =
        if manual_conf_request
          LanSequence()
        else
          log.info("Configured network found: #{network_configured?}")
          network_configured? :auto : LanSequence()
        end
```

Reply:
yes, it is more clear code for me.